tests/NationalInstruments.Analyzers.UnitTests/LongChainDottedInvocationAnalyzerTests.cs

```diff
@@ -0,0 +1,331 @@
+         selectedSoftwareContents.GroupBy(software => software.AliasName)
+            .Select(g => g.First())
+            .Distinct();
+
+        private interface ISoftwareContent
+        {
+            string AliasName { get; }
+        }
+    }
+}
+", (12, 10)
+                }, // inside a property getter
+                new object[]
+                {
+                    @"
+using System.Collections.Generic;
+using System.Linq;
+
+namespace NationalInstruments.Analyzers.UnitTests
+{
+    class Test
+    {
+        void Foo(IEnumerable<ISoftwareContent> softwareContent)
+        {
+            softwareContent.Select(content => content.Children.GroupBy(software => software.AliasName).Select(g => g.First()).Distinct());
+        }
+
+        private interface ISoftwareContent
+        {
+            IEnumerable<ISoftwareContent> Children { get; }
+
+            string AliasName { get; }
+        }
+    }
+}
+", (11, 47)
+                } // inside an argument
+            };
+
+        [Fact]
+        public void NI1017_SingleInvocationInStatement_NoDiagnostic()
+        {
+            var test = new TestFile(@"
+using System.Collections.Generic;
+using System.Linq;
+
+namespace NationalInstruments.Analyzers.UnitTests
+{
+    class Test
+    {
+        void Foo(ISoftwareContent softwareContent)
+        {
+            softwareContent.Child.AliasName.TrimEnd();
+        }
+
+        private interface ISoftwareContent
+        {
+            ISoftwareContent Child { get; }
+
+            string AliasName { get; }
+        }
+    }
+}
+");
+
+            VerifyDiagnostics(test);
+        }
+
+        [Fact]
+        public void NI1017_WellSplitStatementWithDifferentKindsOfInvocations_NoDiagnostic()
+        {
+            var test = new TestFile(@"
+using System.Collections.Generic;
+using System.Linq;
+
+namespace NationalInstruments.Analyzers.UnitTests
+{
+    class Test
+    {
+        string Foo(ISoftwareContent softwareContent) => softwareContent
+            .Child().AliasName.Length
+            .ToString();
+
+        private interface ISoftwareContent
+        {
+            ISoftwareContent Child();
+
+            string AliasName { get; }
+        }
+    }
+}
+");
+
+            VerifyDiagnostics(test);
+        }
+
+        [Theory]
+        [MemberData(nameof(WellSplitStatementInVariousContexts))]
+        public void NI1017_WellSplitStatement_NoDiagnostic(string testString)
+        {
+            var test = new TestFile(testString);
+
+            VerifyDiagnostics(test);
+        }
+
+        [Fact]
+        public void NI1017_PoorlySplitStatementWithDifferentKindsOfInvocations_EmitsDiagnostic()
+        {
+            var test = new TestFile(@"
+using System.Collections.Generic;
+using System.Linq;
+
+namespace NationalInstruments.Analyzers.UnitTests
+{
+    class Test
+    {
+        string Foo(ISoftwareContent softwareContent) => softwareContent.Child().AliasName.Length.ToString();
```

Comment:
Why does `softwareContent.Child().AliasName.Length.ToString()` emit a diagnostic? There's no lambdas in them?

Reply:
[Fixed](https://github.com/ni/csharp/blob/users/tsharma/add-long-dotted-chain-analyzer/tests/NationalInstruments.Analyzers.UnitTests/ChainOfMethodsWithLambdasAnalyzerTests.cs#L221) in the latest update.